symfony/src/Service/SiteSyncService.php

```diff
@@ -0,0 +1,260 @@
+<?php
+
+namespace App\Service;
+
+use App\Entity\Awardee;
+use App\Entity\Organization;
+use App\Entity\Site;
+use Doctrine\ORM\EntityManagerInterface;
+use Pmi\Audit\Log;
+use Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface;
+use Symfony\Component\Serializer\Normalizer\ObjectNormalizer;
+use Symfony\Component\Serializer\Serializer;
+
+class SiteSyncService
+{
+    private $rdrApiService;
+    private $em;
+    private $env;
+    private $loggerService;
+    private $orgEndpoint = 'rdr/v1/Awardee?_inactive=true';
+    private $entries;
+
+    public function __construct(
+        RdrApiService $rdrApiService,
+        EntityManagerInterface $em,
+        EnvironmentService $env,
+        LoggerService $loggerService,
+        ParameterBagInterface $params
+    ) {
+        $this->rdrApiService = $rdrApiService;
+        $this->em = $em;
+        $this->env = $env;
+        $this->loggerService = $loggerService;
+        $this->params = $params;
+    }
+
+    private function getAwardeeEntriesFromRdr()
+    {
+        if (!is_null($this->entries)) {
+            return $this->entries;
+        }
+        $response = $this->rdrApiService->get($this->orgEndpoint);
+        $responseObject = json_decode($response->getBody()->getContents());
+        if ($responseObject && !empty($responseObject->entry)) {
+            $this->entries = $responseObject->entry;
+            return $this->entries;
+        }
+        return [];
+    }
+
+    private function getSitesFromDb()
+    {
+        $sitesRepository = $this->em->getRepository(Site::class);
+        $sites = $sitesRepository->findBy(['deleted' => 0]);
+        $sitesById = [];
+        foreach ($sites as $site) {
+            $sitesById[$site->getGoogleGroup()] = $site;
+        }
+        return $sitesById;
+    }
+
+    private static function getSiteSuffix($site)
+    {
+        return str_replace(\Pmi\Security\User::SITE_PREFIX, '', $site);
+    }
+
+    public function dryRun()
+    {
+        return $this->sync(true);
+    }
+
+    public function sync($preview = false)
+    {
+        $sitesCount = 0;
+        $created = [];
+        $modified = [];
+        $existingSites = $this->getSitesFromDb();
+        $deleted = array_keys($existingSites); // add everything to the deleted array, then remove as we find them
+        $entries = $this->getAwardeeEntriesFromRdr();
+        $serializer = new Serializer([new ObjectNormalizer()]);
```

Comment:
Instead of instantiating a new serializer and having to pass in a normalizer, you can just inject the normalizer into the service constructor:

```php
use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
...
// add $normalizer class variable, add `NormalizerInterface $normalizer` to constructor parameters and assign
...
// then, use it where needed:
$this->normalizer->normalize($existing);
...
```
I tested this and the result is the same as the ObjectNormalizer.